specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/computeRPCommon.json

```diff
@@ -1913,6 +1913,103 @@
         }
       },
       "description": "Specifies additional settings to be applied when patch mode AutomaticByPlatform is selected in Linux patch settings."
+    },
+    "AttachDetachDataDisksRequest": {
+      "properties": {
+        "attachDataDisks": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/definitions/AttachDataDisk"
+          },
+          "description": "The list of managed data disks to be attached."
+        },
+        "detachDataDisks": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/definitions/DetachDataDisk"
+          },
+          "description": "The list of managed data disks to be detached."
+        }
+      },
+      "description": "Specifies the input for attaching and detaching a list of managed data disks."
+    },
+    "AttachDetachDataDisksResponse": {
+      "properties": {
+        "attachedDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AttachedDataDisk"
+          },
+          "description": "The list of managed data disks that were attached."
+        },
+        "detachedDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DetachedDataDisk"
+          },
+          "description": "The list of managed data disks that were detached."
+        }
+      },
+      "description": "Response for attach and detach data disks request."
+    },
+    "AttachDataDisk": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ID of the managed data disk."
+        },
+        "lun": {
+          "type": "integer",
+          "description": "The logical unit number of the data disk. This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM. If not specified, lun would be auto assigned."
+        }
+      },
+      "required": [
+        "id"
+      ],
+      "description": "Describes the data disk to be attached."
+    },
+    "DetachDataDisk": {
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

"id" is used in top-level model, should not be used in nested properties, please change the name.

Reply:
Changed to diskId . Please point out the id in the top-level model for better understanding